### PR TITLE
repair: coroutinize some functions

### PR DESCRIPTION
**Pull request by tchaikov**
also, take this opportunity to let `handle_mutation_fragment()` return void. for better readability.

### CODE REVIEW

**Comment by avikivity** [comment]
@asias please review

**Comment by tchaikov** [comment]
@asias hi Asias, ping for reviews?

**Comment by tchaikov** [comment]
@asias hi Asias, ping for reviews?

File: repair/row_level.cc [hunk 1]
```diff
@@ -1090,69 +1090,58 @@ class repair_meta {
...
+            if (_cmp(r.boundary(), *_current_sync_boundary) <= 0) {
+                break;
+            }
+            _working_row_buf.push_front(std::move(r));
+            co_await coroutine::maybe_yield();
+        }
+        _row_buf.resize(_row_buf.size() - _working_row_buf.size());
+        _row_buf.swap(_working_row_buf);
+        if (sz != _working_row_buf.size() + _row_buf.size()) {
+            co_await coroutine::return_exception(
```
Comment (asias) [inline repair/row_level.cc:1141]: Can't we just call throw here? Why co_await coroutine::return_exception?

**Comment by asias** [comment]
Looks good to me. I have a minor comment inline. 

Btw you can just set the titlle as 

repair: coroutinize some functions

No need to add row_level since it is the only repair we support now;)

File: repair/row_level.cc (hunk 1 above)
Comment (tchaikov) [reply repair/row_level.cc:1141]: good point. i thought this was the recommended approach for no reasons. after a second thought, i realized that `coroutine::return_exception()`  was designed for the case where we want to propagate an existing exception. will change to a plain `throw` statement here.

**Comment by tchaikov** [comment]
> Looks good to me. I have a minor comment inline.
> 
> Btw you can just set the titlle as
> 
> repair: coroutinize some functions
> 
> No need to add row_level since it is the only repair we support now;)

hi Asias, thank you for your review and suggestion. changed the title of this pull request accordingly. ahh, i see now =)

**Comment by tchaikov** [comment]
v2:

- change the `coroutine::return_exception()` call to plain `throw` statement.

**Review by asias — APPROVED** [review]

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hi maintainers, could you help queue this changeset?

**Pull request closed without merging**